### PR TITLE
docs - question in dashboards

### PR DESCRIPTION
**Pull request by jeff-bruemmer**
image

This PR:

- Documents the new ability to save questions in dashboards.
- Reorganizes question docs into the following sections:
  - Questions (the entity). Content about how questions work, what you can do with them:  export results, set up alerts, save to dashboards, etc.
  - Viz
  - Query builder
  	- New pages for filtering and summarizing
  - Native editor
- Relocates public links to embedding.
- Promotes Alerts and Exporting data.
- Splits query builder docs into multiple sections (e.g., filters, summarizing and grouping). Which leaves room for further modularization (e.g., a custom column page).
- Clarifies that you can just copy/paste question links
- Recontextualizes the content in the answers page, and deletes the answers page.
- Removes the answer page.
- Removes the pulses page.
- And some other cleanup.

## Why the reorganization? 

- Establishes questions as a thing in Metabase. The previous organization conflated the concepts of questions and query builder.
- Makes room for more docs on the query builder.
- Elevates viz docs in the hierarchy.

We'll need to update the docs sidebar in the website.

### CODE REVIEW

File: docs/questions/introduction.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+You can also build a new question from an existing question. You won't overwrite the existing question, so feel free to play around. You can use either the [query builder](./query-builder/editor.md) or the [native code editor](./native-editor/referencing-saved-questions-in-queries.md).
+
+Some kinds of saved questions, however, can't be used as source data:
+
+- Druid questions
+- Mongo questions
+- Questions that use `Cumulative Sum` or `Cumulative Count` aggregations
+- Questions that have columns that are named the same or similar thing, like `Count` and `Count 2`
+
+## Saving questions
```
Comment (jeff-bruemmer) [inline docs/questions/introduction.md:43]: Relevant section for saving questions.

File: docs/dashboards/introduction.md [hunk 2]
```diff
@@ -28,10 +28,21 @@ In the top right of the screen, click the **+ New** > **Dashboard**. Give your n
 
 ## Adding questions to a dashboard
 
-There are two ways to add questions to a dashboard: from the dashboard, or from the question you want to add.
+There are different ways to add questions to a dashboard:
 
-- **From a question**: You can add a newly saved question to a dashboard directly from the modal that pops up after you save the question for the first time. You can also add a question to a dashboard by clicking on the ellipsis (**...**) at the top right of a question and selecting **Add to dashboard**.
-- **From a dashboard**: Click on the **pencil** icon to **edit the dashboard**. Then click the **+** icon in the top right of the dashboard editing interface (not the **+** in the main navigation bar) to add any of your saved questions to the dashboard, regardless of which collection the questions are in.
+### Saving questions to a dashboard
```
Comment (jeff-bruemmer) [inline docs/dashboards/introduction.md:33]: Another section on questions in dashboards.

File: docs/dashboards/introduction.md [hunk 3]
```diff
@@ -28,10 +28,21 @@ In the top right of the screen, click the **+ New** > **Dashboard**. Give your n
...
+
+### Adding a question in a collection
+
+Visit a collection and click on the question. Click on the three-dot menu (**...**) at the top right of a question and select **Add to dashboard**.
+
+### From a dashboard
+
+When viewing a dashboard, click on the **pencil** icon to **edit the dashboard**. Then click the **+** icon in the top right of the dashboard editing interface (not the **+** in the main navigation bar).
+
+Keep in mind, you can't add questions that are saved to other dashboards. If you want to use a question in more than one dashboard, you should saved it to a collection. You could also duplicate a question, and save that duplicate to your dashboard.
```
Comment (luizarakaki) [inline docs/dashboards/introduction.md:45]: typo?

"you should save"

File: docs/dashboards/introduction.md (hunk 2 above)
Comment (luizarakaki) [reply docs/dashboards/introduction.md:33]: One of the main changes of this project is that now people can create questions when in the dashboard edit mode.

This section explains how to add a question from a dashboard, but it isn't clear that you can create a new question

**Review by luizarakaki — COMMENTED** [review]
Added a couple comments

There are several complicated interactions with the Trash that are not documented (and I'm not sure entirely sure if we should document, but I guess so).

- Sending a dashboard to the trash, also sends all dashboard questions to the trash.
- Removing a dashboard question from a dashboard sends the question to the trash.
- Restoring a DQ from the trash adds it again to the dashboard.

There are other things like this that would be good to confirm with John.

**Review by abdhabli — APPROVED** [review]

**Pull request merged**